tests/unit/test_httpchecksum.py

```diff
@@ -301,6 +301,34 @@ def test_apply_request_checksum_flex_header_trailer_explicit_digest(self):
         # The body should not have been wrapped
         self.assertIsInstance(request["body"], bytes)
 
+    def test_apply_request_checksum_content_encoding_preset(self):
+        request = self._build_request(b"")
+        request["context"]["checksum"] = {
+            "request_algorithm": {
+                "in": "trailer",
+                "algorithm": "crc32",
+                "name": "x-amz-checksum-crc32",
+            }
+        }
+        request["headers"]["Content-Encoding"] = "foo"
+        apply_request_checksum(request)
+        # The content encoding should not have been modified
```

Comment:
This may be more accurate. We are modifying it, but we don't want to overwrite it.
```suggestion
        # The content encoding should only have been appended
```